planner/core/point_get_plan.go

```diff
@@ -488,36 +488,42 @@ func tryUpdatePointPlan(ctx sessionctx.Context, updateStmt *ast.UpdateStmt) Plan
 				HandleOrdinal: handleCol.Index,
 			},
 		},
+		NotAllConstant: notAllConst,
 	}.Init(ctx)
 	updatePlan.SetSchema(fastSelect.schema)
 	return updatePlan
 }
 
-func buildOrderedList(ctx sessionctx.Context, fastSelect *PointGetPlan, list []*ast.Assignment) []*expression.Assignment {
+func buildOrderedList(ctx sessionctx.Context, fastSelect *PointGetPlan, list []*ast.Assignment) ([]*expression.Assignment, bool) {
```

Comment:
how about refactoring the function signature to:
```go
func buildOrderedList(ctx sessionctx.Context, fastSelect *PointGetPlan, list []*ast.Assignment) (
	orderedList []*expression.Assignment,
	notAllConstant bool,
) {
```
give each return item a name can help readers understand the code.